Automattic/wp-calypso
# Assembler First Flow: Handle the launch site task

Related to https://github.com/Automattic/wp-calypso/issues/83390

## Proposed Changes

* Check whether the site is able to be launched
* Go to the checkout if the user selects a paid plan or a paid domain before launching the site
* Launch the site after the user clicks the “Launch your site” button in the launchpad
* Navigate to the Celebration step after launching 

### Demo

https://github.com/Automattic/wp-calypso/assets/13596067/9d833c1d-4985-4b82-90f4-f010eb114170

### The Celebration step

![image](https://github.com/Automattic/wp-calypso/assets/13596067/c1b11835-ed18-4c5a-8b65-5d007ee1b4a2)

## Testing Instructions



* Apply latest jetpack-mu-wpcom to your sandbox
  ```
  bin/jetpack-downloader test jetpack-mu-wpcom-plugin trunk
  ```
* Go to `/setup/assembler-first?flags=themes/assembler-first`
* Create a new site
* Finish the Assembler step
* Go to the launchpad, `/setup/assembler-first/launchpad?siteSlug=<your_site>`
* Select a plan
* Select a domain
* Click the “Launch your site” button
* Ensure you can see the Celebration step after launching
* Reset jetpack-mu-wpcom on your sandbox
  ```
  jetpack-downloader reset jetpack-mu-wpcom-plugin
  ```

## Pre-merge Checklist
